rllib/utils/test_utils.py

```diff
@@ -497,33 +507,55 @@ def run_learning_tests_from_yaml(
                 "evaluation_interval", None) is not None
 
             # Error: Increase failure count and repeat.
-            if t.status == "ERROR":
+            if any(t.status == "ERROR" for t in trials_for_experiment):
+                print(f" ... ERROR.")
                 checks[experiment]["failures"] += 1
             # Smoke-tests always succeed.
             elif smoke_test:
+                print(f" ... SMOKE TEST (mark ok).")
                 checks[experiment]["passed"] = True
                 del experiments_to_run[experiment]
             # Experiment finished: Check reward achieved and timesteps done
             # (throughput).
             else:
-                reward_mean = \
-                    t.last_result["evaluation"]["episode_reward_mean"] if \
-                    check_eval else t.last_result["episode_reward_mean"]
+                if check_eval:
```

Comment:
totally minor, but maybe break everything in this else clause to a _private util fun _check_result(check_eval).
so it's clear where to update in case we add more conditions.